### PR TITLE
Fix Codacy warnings

### PR DESCRIPTION
**Pull request by MarcelRaad**
Mostly from CppCheck. Some fixes for unused local variable warnings would have been nicer with C99, being able to declare them in the right preprocessor condition block.

### CODE REVIEW

**Comment by MarcelRaad** [comment]
Is there any area keyword I can use in the commit titles when touching everything, including even the packages directory? I haven't found one consistently used.

**Comment by jay** [comment]
> I haven't found one consistently used.

my opinion - use `build: ` and squash

~~~
build: fix codacy warnings

- Consistently use WIN32 macros in Windows.

- Fix unused local variable warnings.

Some variables are only used conditionally, so also declare them conditionally.

- Fix no-consecutive-blank-lines Codacy warning.

- Suppress unused variable warnings.

- etc

Closes #xxxx
~~~

File: lib/if2ip.c [hunk 1]
```diff
@@ -149,9 +151,10 @@ if2ip_result_t Curl_if2ip(int af, unsigned int remote_scope,
 
                 continue;
               }
+
+              if (scopeid)
```
Comment (MarcelRaad) [inline lib/if2ip.c]: No idea how that happaned. Probably my Visual Studio forgot that if goes without a space when moving the block.

**Comment by MarcelRaad** [comment]
> my opinion - use build: and squash

Thanks @jay , makes sense!

File: lib/if2ip.c (hunk 1 above)
Comment (MarcelRaad) [reply lib/if2ip.c]: Fixed.

**Pull request closed without merging**